### PR TITLE
add full types for RoomVisual

### PR DESCRIPTION
**Pull request by Jomik**
Adds interfaces for all style options and also helpers in `utils.Options`.

I do propose moving the `RoomVisual.*Style` interfaces into an `api.Options` file, together with other option interfaces.

### CODE REVIEW

File: src/main/kotlin/screeps/utils/Options.kt [hunk 1]
```diff
@@ -1,7 +1,73 @@
...
 import screeps.api.Game
+import screeps.api.LineStyleConstant
+import screeps.api.RoomVisual
+import screeps.api.TextAlignConstant
 
 fun routeOptions(routeCallback: (roomName: String, fromRoomName: String) -> Double): Game.Map.RouteOptions = jsObject {
     this.routeCallback = routeCallback
 }
+
+// RoomVisual style options
```
Comment (exaV) [inline src/main/kotlin/screeps/utils/Options.kt]: This comment already hints at the fact that the functions below are only used with RoomVisual.
Please put them into the file screeps.api.RoomVisual.kt

As discussed in https://github.com/exaV/screeps-kotlin-types/pull/11 we should put things  that belong together into the same file.

File: src/main/kotlin/screeps/api/Constants.kt [hunk 2]
```diff
@@ -263,8 +263,19 @@ external val ORDER_BUY: OrderConstant
 external val SUBSCRIPTION_TOKEN: TradableConstant
 
 // Constants to restrict function arguments
```
Comment (exaV) [inline src/main/kotlin/screeps/api/Constants.kt]: Could you change this comment to something that better reflects that the declarations below are constants that are used as such but never declared in the api.

File: src/main/kotlin/screeps/utils/Options.kt (hunk 1 above)
Comment (exaV) [reply src/main/kotlin/screeps/utils/Options.kt]: I am aware that there are already some deviations from this throughout the api (e.g. the file Utils.kt), we will fix  those as before 1.0 just not in this PR.

File: src/main/kotlin/screeps/api/Constants.kt (hunk 2 above)
Comment (Jomik) [reply src/main/kotlin/screeps/api/Constants.kt]: Tried..

**Pull request merged**